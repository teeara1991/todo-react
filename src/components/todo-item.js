import React from "react";
export function TodoItem(props) {
  return (
    <ul className={`todo-item ${props.completed ? "completed" : ""}`}>
      <input
        type="checkbox"
        checked={props.checked}
        onChange={props.onComplete}
      />
      <span className="todo-text" onClick={props.onComplete}>
        {props.name}
      </span>
      <span className="todo-trash">
        <i className="fas fa-trash-alt" onClick={props.onDelete}></i>
      </span>
    </ul>
  );
}
