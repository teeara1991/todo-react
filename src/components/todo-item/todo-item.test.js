import React from "react";
import { TodoItem} from "../todo-item/todo-item";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() });
it('matches the snapshot', () => {
    const props = {
       name: 'Buy groceries', checked:false, completed: false
      };
    const tree = shallow(<TodoItem {...props}/>)
    expect(toJson(tree)).toMatchSnapshot()
  })