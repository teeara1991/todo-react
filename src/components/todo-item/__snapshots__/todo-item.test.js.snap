// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<ul
  className="todo-item "
>
  <input
    checked={false}
    type="checkbox"
  />
  <span
    className="todo-text"
  >
    Buy groceries
  </span>
  <span
    className="todo-trash"
  >
    <i
      className="fas fa-trash-alt"
    />
  </span>
</ul>
`;
